store/driver/txn/union_iter.go

```diff
@@ -78,6 +81,13 @@ func (iter *UnionIter) updateCur() error {
 
 		if !iter.snapshotValid {
 			iter.curIsDirty = true
+			// if delete it
+			if len(iter.dirtyIt.Value()) == 0 {
+				if err := iter.dirtyNext(); err != nil {
+					return err
+				}
+				continue
+			}
```

Comment:
is it a bug fix?

Reply:
Not a bug fix. First I introduce `UnionIter` is here:  #26353 . The `iterTxnMemBuffer` will filter out empty value itself, so it is not necessary to skip empty values in dirtyIt (Both behavior is right). But for current pr, I think its better to keep it the some behavior with the on in client-go because it is used in transaction now. So I copied the codes again from client-go to here.